UndertaleModTool/Scripts/Community Scripts/ImportGMS2FontData.csx

```diff
@@ -38,23 +38,38 @@ string textureSourcePath = Path.Combine(Path.GetDirectoryName(importFile), fontN
 if (!File.Exists(textureSourcePath))
     throw new ScriptException("The font texture file " + textureSourcePath + " doesn't exist.");
 
+UndertaleTextureGroupInfo defaultTexGroup = null;
+bool disappearMigitation = Data.GM2022_3;
+if (disappearMigitation)
+{
+    defaultTexGroup = Data.TextureGroupInfo.ByName("Default");
+    if (defaultTexGroup == null)
+        throw new ScriptException("Default texture group doesn't exist.");
+}
+
 UndertaleFont font = Data.Fonts.ByName(fontName);
 if (font == null)
 {
-    font = new UndertaleFont() {
+    font = new UndertaleFont()
+    {
         Name = Data.Strings.MakeString(fontName)
     };
     Data.Fonts.Add(font);
+    if (disappearMigitation)
+        defaultTexGroup.Fonts.Add(new UndertaleResourceById<UndertaleFont, UndertaleChunkFONT>() { Resource = font });
 }
 
 Bitmap textureBitmap = new Bitmap(textureSourcePath);
+textureBitmap.SetResolution(96.0F, 96.0F); // dpi fix
 UndertaleEmbeddedTexture texture = new UndertaleEmbeddedTexture();
-texture.Name = Data.Strings.MakeString("Texture " + Data.EmbeddedTextures.Count); // ???
+texture.Name = new UndertaleString("Texture " + Data.EmbeddedTextures.Count); // ???
```

Comment:
Why do you create a new UndertaleString instead of using MakeString?

Reply:
Because every other script did that, IDK why a UndertaleString was used anyway

Reply:
You should probably revert that then, as MakeString is safer.

Reply:
I mean the other scripts don't add the new string to Data.Strings like MakeString does, I don't think anything requires the "name" to be actually in data.win too

Reply:
Ah wait, it's for the texture name. Yeah, carry on then.